chain/network/src/peer_manager.rs

```diff
@@ -726,8 +754,18 @@ impl PeerManagerActor {
         self.routing_table.add_ping(ping);
     }
 
+    /// Handle pong messages. Add pong temporary to the routing table, mostly used for testing.
+    /// If `metric_recorder` feature flag is enabled, save how much time passed since we sent ping.
     fn handle_pong(&mut self, _ctx: &mut Context<Self>, pong: Pong) {
-        self.routing_table.add_pong(pong);
+        #[cfg(feature = "metric_recorder")]
```

Comment:
I don't think this will work this way. They feature you are probably talking about is:

`if !cfg(metric_recorder) { ... } else { ... }`

But in practice what the compiler does is substituting by true/false this `!cfg(metric_recorder)` without pruning the unreachable branch (at least in the first pass), so it won't compile if some fields are compiled only behind the flag, which is the case here. 

I know that code legibility degraded notably in that function, but it was the best I managed to put together.

Reply:
Okay I see.